plugins/scaffolder-backend/src/scaffolder/actions/builtin/catalog/fetchMultiple.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CatalogApi } from '@backstage/catalog-client';
+import { createTemplateAction } from '@backstage/plugin-scaffolder-node';
+import yaml from 'yaml';
+
+const id = 'catalog:fetchMultiple';
+
+const examples = [
+  {
+    description: 'Fetch entities by reference',
+    example: yaml.stringify({
+      steps: [
+        {
+          action: id,
+          id: 'fetchMultiple',
+          name: 'Fetch catalog entities',
+          input: {
+            entityRefs: ['component:default/name', 'component:default/name2'],
+            fields: ['kind', 'metadata.name'],
+          },
+        },
+      ],
+    }),
+  },
+];
+
+/**
+ * Returns entities from the catalog by entity references.
+ * @public
+ */
+export function createFetchCatalogEntitiesAction(options: {
+  catalogClient: CatalogApi;
+}) {
+  const { catalogClient } = options;
+
+  return createTemplateAction<{ entityRefs: string[]; fields?: string[] }>({
+    id,
+    description: 'Returns entities from the catalog by entity references',
+    examples,
+    schema: {
+      input: {
+        required: ['entityRefs'],
+        type: 'object',
+        properties: {
+          entityRefs: {
+            type: 'array',
+            title: 'Entity references',
+            description: 'Entity references of the entities to get',
+          },
+          fields: {
+            type: 'array',
+            title: 'Entity fields',
+            description: 'Entity fields to return for each entity',
+          },
+        },
+      },
+      output: {
+        type: 'array',
+        properties: {
+          entity: {
+            title: 'Entities found by the entity reference',
+            type: 'array',
+            description:
+              'Array containing objects with same values used in the Entity schema.',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const { entityRefs, fields } = ctx.input;
+      const entities = await catalogClient.getEntitiesByRefs(
+        { entityRefs, fields },
+        {
+          token: ctx.secrets?.backstageToken,
+        },
+      );
+
+      ctx.output(
+        'entities',
+        entities.items.map(e => (e ? e : null)),
```

Comment:
`e ?? null` could be nice here, or it should be possible to just skip since it's gonna be `null` under the hood anyway, and `undefined` serializes to `null` in arrays in JSON too